### PR TITLE
feat: document how to combine system and application updates

### PR DESCRIPTION
**Pull request by tranchitella**
Ticket: MEN-6042
Changelog: Title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Review by kacf — APPROVED** [review]
Good text! 👍

Some small remarks, but pre-approving.

File: 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+For this reason, it is common to add a dedicated, separated persistent partition to store files that must survive across full system updates. These files may include network parameters, user configuration changes, data collected and processed by the application software on the device, etc. By convention, we call this partition the *data partition*.
+
+
+### Application updates
+
+You can extend the Mender Client supporting different application-specific update patterns using the [Update Module](../08.Create-a-custom-Update-Module/docs.md) framework.
+
+When processing an application update, the Mender Client can write files in any partition available on the device, including the system and the data partitions. However, as mentioned in the System updates section above, any changes to the system partition will be lost on system updates as the device will switch from the active to the inactive partition.
+
+For this reason, if you designed the application update to persist across system updates, you need to ensure that all the files part of the update are installed in any partition but the active (and inactive) ones, for example, using the data partition.
```
Comment (TheMeaningfulEngineer) [inline 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md:27]:
```suggestion
For this reason, if you designed the application update to persist across system updates, you need to ensure that all 
the content of application update (the files the update consist of) end up installed on the *data partition* and **not** on either of the two A/B partitions used for system updates.
```

File: 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+title: Combining system and application updates
+taxonomy:
+    category: docs
+    label: tutorial
+---
+
+There are many possible strategies to combine the system and application updates you deploy to your devices. This section summarizes the crucial points you should be aware of when defining your OTA strategy and outlines the best practices we suggest for common scenarios.
+
+## Basic concepts
+
```
Comment (TheMeaningfulEngineer) [inline 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md:11]: I usually talk about this in presales

```suggestion
### Drawbacks of combining system and application updates

Combining system and application updates gives you the flexibility to separately deploy multiple different software versions on your device. At the same time, there is increase in complexity of your production system as the software on the device is no longer represented by a single software version.

For example your device software is no longer `v.2.5` instead it is `system v.2.5.0` - app1 `v.1.2.4` - app2 `v2.3.1`.

The consequence of this complexity can contribute to:
- longer testing processes
- stricter API versioning and development limitations
- more complexity in customer communication (`What version are you running?`)

There isn't a silver bullet for the update approach and knowing all the pros and cons allows for making an educated decision based on your use case.
```

**Review by TheMeaningfulEngineer — COMMENTED** [review]
Look good!
FWIW
Added a comment on the drawbacks.

Didn't have a change to try the code out though, can't commit to that, will do it if I find time (don't be blocked by this)

**Review by kjaskiewiczz — APPROVED** [review]

**Review by TheYoctoJester — APPROVED** [review]
All in all fine with me. One thing that would add value to the user is a description of how such an example application could be created, or proving example tarballs and matching update modules so the steps can easily replicated.

File: 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md (hunk 2 above)
Comment (TheMeaningfulEngineer) [reply 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md:11]: @TheYoctoJester 
Thanks, killed the bullet.
Like the `combinatorial complexity` term.

Comment (TheMeaningfulEngineer) [reply 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md:11]: @TheYoctoJester 
> optimally with an example on how to provide all involved software versions through the inventory

Not sure what you mean by this though...

Comment (TheYoctoJester) [reply 06.Artifact-creation/03.Combining-system-and-application-updates/docs.md:11]: Okay, I just remember this being a common source of confusion to users when they look at the devices list.

**Review by TheMeaningfulEngineer — APPROVED** [review]
Looks good thanks!

**Pull request merged**